src/components/Modals/ContactCardModal.jsx

```diff
@@ -1,96 +1,85 @@
+  ] = useState(false)
 
-  toggleConfirmDeleteModal = () => {
-    this.setState(state => ({
-      shouldDisplayConfirmDeleteModal: !state.shouldDisplayConfirmDeleteModal
-    }))
+  const toggleConfirmDeleteModal = () => {
+    setShouldDisplayConfirmDeleteModal(!shouldDisplayConfirmDeleteModal)
   }
 
-  deleteContact = async (contactParam = null) => {
-    const { contact, deleteContact, onDeleteContact, onClose } = this.props
+  const deleteContact = async (contactParam = null) => {
+    const { contact, deleteContact, onDeleteContact, onClose } = props
     onClose && onClose()
     await deleteContact(contactParam ? contactParam : contact)
     onDeleteContact && onDeleteContact(contactParam ? contactParam : contact)
   }
 
-  toggleEditMode = () => {
-    this.setState(state => ({
-      editMode: !state.editMode
-    }))
+  const toggleEditMode = () => {
+    setEditMode(!editMode)
   }
 
-  render() {
-    const { onClose, t, id } = this.props
-    const { editMode, shouldDisplayConfirmDeleteModal } = this.state
+  const { onClose, t, id } = props
 
-    return (
-      <Modal into="body" dismissAction={onClose} size="xlarge" mobileFullscreen>
-        <Query query={client => client.get(DOCTYPE_CONTACTS, id)}>
-          {({ data: contact, fetchStatus: fetchContactStatus }) => {
-            return (
-              <Query
-                query={client =>
-                  client
-                    .find('io.cozy.contacts.groups')
-                    .where({
-                      trashed: { $exists: false }
-                    })
-                    .sortBy([{ name: 'asc' }])
-                    .indexFields(['name'])
-                }
-              >
-                {({ data: allGroups, fetchStatus: allGroupsContactStatus }) => {
-                  if (
-                    fetchContactStatus !== 'loaded' ||
-                    allGroupsContactStatus !== 'loaded'
-                  ) {
-                    return <SpinnerContact size="xxlarge" />
-                  }
-                  return (
-                    <DumbContactCardModal
-                      editMode={editMode}
-                      contact={contact}
-                      allGroups={allGroups}
-                      t={t}
-                      toggleConfirmDeleteModal={this.toggleConfirmDeleteModal}
-                      toggleEditMode={this.toggleEditMode}
-                      shouldDisplayConfirmDeleteModal={
-                        shouldDisplayConfirmDeleteModal
-                      }
-                      deleteContact={this.deleteContact}
-                    />
-                  )
-                }}
-              </Query>
-            )
-          }}
-        </Query>
-      </Modal>
-    )
-  }
+  const resultContactById = useQuery(
+    queryContactById(id).definition,
+    queryContactById(id).options
+  )
+  const resultAllGroups = useQuery(
+    queryAllGroups.definition,
+    queryAllGroups.options
+  )
+
+  const showResult =
```

Comment:
nit: I don't like having this kind of naming. It doesn't tell us anything. 

const dataHaveBeenLoaded or something similar is way better IMHO.